strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/NugetArtifactCoordinates.java

```diff
@@ -1,23 +1,78 @@
 package org.carlspring.strongbox.artifact.coordinates;
 
+import java.net.URI;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
 import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlRootElement;
 
 /**
  * @author Sergey Bespalov
  *
  */
-public abstract class NugetArtifactCoordinates extends AbstractArtifactCoordinates
-{
 
+@XmlRootElement(name = "nugetArtifactCoordinates")
+@XmlAccessorType(XmlAccessType.NONE)
+public class NugetArtifactCoordinates extends AbstractArtifactCoordinates
+{
     public static final String ID = "id";
     public static final String VERSION = "version";
     public static final String EXTENSION = "extension";
+    private static final String NUGET_PACKAGE_REGEXP_PATTERN = "([a-zA-Z0-9_.-]+)[/|\\\\]([a-zA-Z0-9_.-]+)[/|\\\\]([a-zA-Z0-9_.-]+).(nupkg|nuspec|nupkg\\.sha512)";
+
 
     public NugetArtifactCoordinates()
     {
         defineCoordinates(ID, VERSION, EXTENSION);
     }
 
+    public NugetArtifactCoordinates(String path)
+    {
+        Pattern pattern = Pattern.compile(NUGET_PACKAGE_REGEXP_PATTERN);
+        Matcher matcher = pattern.matcher(path);
+        if (!matcher.matches())
+        {
+            return;
+        }
+
+        String packageId = matcher.group(1);
+        String version = matcher.group(2);
+        String packageArtifactName = matcher.group(3);
+        String packageArtifactType = matcher.group(4);
+
+        if (!String.format("%s.%s", packageId, version).startsWith(packageArtifactName))
+        {
+            return;
+        }
+
+        setId(packageId);
+        setVersion(version);
+        setType(packageArtifactType);
+    }
+
+    public String toPath()
```

Comment:
@sanket407 can you please move this method below the constructors (better to the end of the class definition)